src/libsystemd-network/test-dhcp6-client.c

```diff
@@ -78,8 +78,6 @@ static int test_ifindex = 42;
 static unsigned test_client_sent_message_count = 0;
 static sd_dhcp6_client *client_ref = NULL;
 
-STATIC_DESTRUCTOR_REGISTER(client_ref, sd_dhcp6_client_unrefp);
```

Comment:
This means that main() doesn't call `static_destruct()`, which it should. This whole test (sh/c)ould be converted to TEST macros. I guess I missed the stray ones that aren't located in `test/`.

Reply:
Yeah, converting the tests to use your macro is another choice to fix the issue. But, I'd like to leave it to later PR.